### ISSUE TITLE
REGEX REPLACE sintax error causing invalid sequence 

### ISSUE THREAD

**Issue by CarlottaSartore** [robotology/yarp#2107]
**Issue Description**

In the file _cmake/YarpIDL.cmake_ line 171 (**devel branch**) 

https://github.com/robotology/yarp/blob/a656496b49d3c4afafef10e8a647abfb39a1e686/cmake/YarpIDL.cmake#L170-L172

causes the installation of a _YarpIDL.cmake_ file  that arises the following error when trying to compile other repositories that depend on Yarp package 

`when parsing string ^.+/([^\/]+)$ Invalid escape sequence \/ `

Solved by modifying the cmake file, hence substituting the aforementioned line with; 

```
      if(NOT _cpp_placement_ns)
        string(REGEX REPLACE "^.+/([^/]+)$" "\\1" _dest_file "${_dest_file}")
      endif()
```
and reinstall the header. 

CC @nunoguedelha

**Comment by drdanz** [comment]
@CarlottaSartore Thanks for debugging and taking time to open this issue.
Would you like to try and create a pull request to fix this issue?

**Comment by drdanz** [comment]
Fixed by #2110
